bin/onlyswaps-solver/src/executor.rs

```diff
@@ -57,61 +60,71 @@ impl<'a, P: Provider> TradeExecutor<'a, P> {
                 .find(|contract| contract.address() == &trade.token_out_addr)
                 .expect("somehow didn't have a token contract binding for a solved trade");
 
-            // in theory, we shouldn't need to wait until the next block because txs will be processed in nonce order
-            // but for whatever reason this doesn't seem to be the case :(
-            let approve: anyhow::Result<TxHash> = async {
-                let tx = token
-                    .approve(*router.address(), trade.swap_amount)
-                    .send()
-                    .await?;
-                let receipt = tx.watch().await?;
-                Ok(receipt)
-            }
-            .await;
-            match approve {
-                Ok(_) => {}
-                Err(e) => {
-                    tracing::error!(e = ?e,"error approving trade");
+            // and finally execute the trade with a timeout
+            match timeout(
+                Duration::from_secs(10),
+                execute_trade(&trade, router, token, self.own_address),
+            )
+            .await
```

Comment:
good idea to add the timeout at this level directly, don't have to individually instrument transactions.